Intersect.Server/Entities/Projectile.cs

```diff
@@ -122,109 +118,205 @@ private void AddProjectileSpawns(List<KeyValuePair<Guid, int>> spawnDeaths)
+                            return Direction.Right;
+                        case Direction.DownRight:
+                            return Direction.Down;
+                        case Direction.DownLeft:
+                            return Direction.Left;
                         default:
                             return projectionDir;
                     }
-                case 3: //Right
+                case Direction.DownLeft:
                     switch (projectionDir)
                     {
-                        case 0: //Up
-                            return 3;
-                        case 1: //Down
-                            return 2;
-                        case 2: //Left
-                            return 0;
-                        case 3: //Right
-                            return 1;
-                        case 4: //UpLeft
-                            return 5;
-                        case 5: //UpRight
-                            return 7;
-                        case 6: //DownLeft
-                            return 4;
-                        case 7: //DownRight
-                            return 6;
+                        case Direction.Up:
+                            return Direction.DownLeft;
+                        case Direction.Down:
+                            return Direction.UpRight;
+                        case Direction.Left:
+                            return Direction.DownRight;
+                        case Direction.Right:
+                            return Direction.UpLeft;
+                        case Direction.UpLeft:
+                            return Direction.Down;
+                        case Direction.UpRight:
+                            return Direction.Left;
+                        case Direction.DownRight:
+                            return Direction.Up;
+                        case Direction.DownLeft:
+                            return Direction.Right;
+                        default:
+                            return projectionDir;
+                    }
+                case Direction.DownRight:
+                    switch (projectionDir)
+                    {
+                        case Direction.Up:
+                            return Direction.DownRight;
+                        case Direction.Down:
+                            return Direction.UpLeft;
+                        case Direction.Left:
+                            return Direction.UpRight;
+                        case Direction.Right:
+                            return Direction.DownLeft;
+                        case Direction.UpLeft:
+                            return Direction.Right;
+                        case Direction.UpRight:
+                            return Direction.Down;
+                        case Direction.DownRight:
+                            return Direction.Left;
+                        case Direction.DownLeft:
+                            return Direction.Up;
```

Comment:
My refactor suggestion for this on the client applies here too. This can be extract into a common utility method inside of Core, these switch cases that need to be identical are far too large (aka error prone)

Reply:
https://github.com/AscensionGameDev/Intersect-Engine/commit/fddf6be4dc2cca42dd1afefaa8e9fd1ca3b87a7b